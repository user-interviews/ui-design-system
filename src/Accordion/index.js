import Accordion from './Accordion';
import AccordionItem from './AccordionItem';
import AccordionButton from './AccordionButton';
import AccordionHeader from './AccordionHeader';
import AccordionBody from './AccordionBody';
import AccordionToggle from './AccordionToggle';
import AccordionCollapse from './AccordionCollapse';

export {
  Accordion,
  AccordionItem,
  AccordionButton,
  AccordionHeader,
  AccordionBody,
  AccordionToggle,
  AccordionCollapse,
};
