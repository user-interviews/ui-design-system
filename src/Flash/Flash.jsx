import React from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup } from 'react-transition-group';
import classNames from 'classnames';

import FadeTransition from 'src/FadeTransition';
import AlertMessage from './AlertMessage';

import './Flash.scss';

export default function Flash(props) {
  const groupClassNames = classNames(
    'container',
    'Flash',
    { 'Flash--no-header': !props.header },
  );

  return (
    <TransitionGroup className={groupClassNames}>
      {
        props.messages.map((message) => (
          <FadeTransition key={message.id}>
            <AlertMessage
              autoDismiss={props.autoDismiss}
              id={message.id}
              message={message.message}
              type={message.type}
              onDismiss={props.onFlashClosed}
            />
          </FadeTransition>
        ))
      }
    </TransitionGroup>
  );
}

Flash.propTypes = {
  autoDismiss: PropTypes.bool,
  header: PropTypes.bool,
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      message: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
      type: PropTypes.string,
    }),
  ).isRequired,
  onFlashClosed: PropTypes.func,
};

Flash.defaultProps = {
  autoDismiss: false,
  header: true,
  onFlashClosed: undefined,
};
