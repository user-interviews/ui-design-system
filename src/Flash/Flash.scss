@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '~bootstrap/scss/alert';
@import '../../scss/theme';

$top-padding: 1.25rem;
// Flash colors
$ux-flash-error: $ux-red-100;
$ux-flash-error-border: $ux-red-200;

$ux-navbar-shadow-color: rgba( 0, 0, 0, 0.26 );
$ux-box-shadow: 0 2px 5px $ux-navbar-shadow-color;
$ux-box-shadow-light: 0 1px 1px $ux-navbar-shadow-color;

$ux-box-shadow-card: 0 2px 4px rgba(0,0,0,0.1);

$ux-box-shadow-top: 0 -2px 5px $ux-navbar-shadow-color;
$ux-box-shadow-top-light: 0 -1px 1px $ux-navbar-shadow-color;

// This zindex is between the header (zindex-fixed) and the modal backdrop (zindex-modal-backdrop)
$z-index-flash: 1045;

.flash {
  position: fixed;
  top: $top-padding + $ui-navbar-height;
  left: 50%;
  transform: translateX(-50%);

  z-index: $z-index-flash;

  @include media-breakpoint-down(sm) {
    top: $top-padding + $ui-navbar-height-mobile;
  }

  &--no-header {
    top: $top-padding;
  }
}

.alert {
  font-weight: 500;
}

.alert-info {
  @extend .alert-warning;
}

.alert-error {
  @extend .alert-danger;

  background-color: $ux-flash-error;
  border-color: $ux-flash-error-border;
}
