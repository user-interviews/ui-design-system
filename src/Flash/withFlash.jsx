import React from 'react';
import PropTypes from 'prop-types';

import Flash from './Flash';
import useFlash from './useFlash';

export const withFlashPropTypes = {
  setFlashMessage: PropTypes.func,
};

export default function withFlash(WrappedComponent) {
  return (props) => {
    const { messages, setMessage, dismissMessage } = useFlash();

    return (
      <>
        <Flash
          messages={messages}
          onFlashClosed={dismissMessage}
        />
        <WrappedComponent
          setFlashMessage={setMessage}
          /* eslint-disable-next-line react/jsx-props-no-spreading */
          {...props}
        />
      </>
    );
  };
}
