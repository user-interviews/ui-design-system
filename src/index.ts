import {
  Accordion,
  AccordionCollapse,
  AccordionItem,
  AccordionToggle,
} from './Accordion';
import { Alert, MessageTypes } from './Alert';
import Button, { ButtonSizes, ButtonVariants } from './Button';
import Avatar from './Avatar';
import Card, { CardSizes } from './Card';
import { CardStack } from './CardStack';
import CheckboxButton, { CHECKED_STATES } from './CheckboxButton';
import CheckboxButtonGroup from './CheckboxButtonGroup';
import { Col, Container, Row } from './Container';
import { ORIENTATIONS as BUTTON_GROUP_ORIENTATIONS } from './ControlButtonGroup';
import CopyToClipboard from './CopyToClipboard';
import CopyToClipboardButton from './CopyToClipboardButton';
import {
  Drawer,
  DrawerHeader,
  DrawerBody,
  DrawerFooter,
  DrawerSizes,
} from './Drawer';
import {
  Dropdown,
  DropdownDivider,
  DropdownToggle,
  DropdownItem,
  DropdownMenu,
} from './Dropdown';
import EmptyState from './EmptyState';
import FadeTransition from './FadeTransition';
import { Flex } from './Flex';
import Form from './Form';
import FormControlLabel from './FormControlLabel';
import FormGroup from './FormGroup';
import { Heading } from './Heading';
import IconButton from './IconButton';
import IconCell from './IconCell';
import Input from './Input';
import InputLabel from './InputLabel';
import InputLegend from './InputLegend';
import { LoadingSkeleton } from './LoadingSkeleton';
import LoadingOverlay from './LoadingOverlay';
import Main from './Main';
import {
  Modal,
  MODAL_SIZES,
  ModalBody,
  ModalFooter,
  ModalHeader,
} from './Modal';
import MoneyInput from './MoneyInput';
import { OverlayTrigger, OVERLAY_TRIGGER_PLACEMENT } from './OverlayTrigger';
import { Pill, Pills, PILL_COLORS } from './Pill';
import { Popover, PopoverBody, PopoverCard } from './Popover';
import Popper from './Popper';
import ProfileCell, { ProfileCellSkeleton } from './ProfileCell';
import RadioButton from './RadioButton';
import RadioButtonGroup from './RadioButtonGroup';
import {
  RichTextEditor,
  RichTextEditorActions,
  RichTextEditorAllActionsArray,
  RichTextEditorDefaultActionsArray,
} from './RichTextEditor';
import {
  AsyncSelect,
  AsyncCreatableSelect,
  CreatableSelect,
  Option,
  OptionWithDescription,
  SELECT_SIZES,
  SelectComponents,
  SingleSelect,
  ValueContainer,
} from './Select';
import { colors as COLORS } from './Styles';
import { Step, Steps } from './Steps';
import {
  Tab,
  Tabs,
} from './Tabs';
import {
  Table,
  TableBody,
  TableCell,
  TableFoot,
  TableHead,
  TableLoadingSkeleton,
  TableRow,
  TableSortLabel,
} from './Table';
import { Text, TEXT_PROPS } from './Text';
import {
  Toast,
  useToast,
  withToast,
  withToastPropTypes,
} from './Toast';
import { ToggleInput } from './ToggleInput';
import Tooltip from './Tooltip';

export {
  Accordion,
  AccordionCollapse,
  AccordionItem,
  AccordionToggle,
  Alert,
  Avatar,
  AsyncSelect,
  AsyncCreatableSelect,
  Button,
  ButtonSizes,
  ButtonVariants,
  BUTTON_GROUP_ORIENTATIONS,
  Card,
  CardSizes,
  CardStack,
  CheckboxButton,
  CheckboxButtonGroup,
  CHECKED_STATES,
  Col,
  COLORS,
  Container,
  CopyToClipboard,
  CopyToClipboardButton,
  CreatableSelect,
  Drawer,
  DrawerHeader,
  DrawerBody,
  DrawerFooter,
  DrawerSizes,
  Dropdown,
  DropdownDivider,
  DropdownToggle,
  DropdownItem,
  DropdownMenu,
  EmptyState,
  FadeTransition,
  Flex,
  Form,
  FormControlLabel,
  FormGroup,
  Heading,
  IconButton,
  IconCell,
  Input,
  InputLabel,
  InputLegend,
  LoadingOverlay,
  LoadingSkeleton,
  Main,
  MessageTypes,
  Modal,
  MODAL_SIZES,
  ModalBody,
  ModalFooter,
  ModalHeader,
  MoneyInput,
  Option,
  OptionWithDescription,
  OverlayTrigger,
  OVERLAY_TRIGGER_PLACEMENT,
  Pill,
  Pills,
  PILL_COLORS,
  Popover,
  PopoverBody,
  PopoverCard,
  Popper,
  ProfileCell,
  ProfileCellSkeleton,
  RadioButton,
  RadioButtonGroup,
  RichTextEditor,
  RichTextEditorActions,
  RichTextEditorAllActionsArray,
  RichTextEditorDefaultActionsArray,
  Row,
  SELECT_SIZES,
  SelectComponents,
  SingleSelect,
  Step,
  Steps,
  Tab,
  Tabs,
  Table,
  TableBody,
  TableCell,
  TableFoot,
  TableHead,
  TableLoadingSkeleton,
  TableRow,
  TableSortLabel,
  Text,
  TEXT_PROPS,
  Toast,
  ToggleInput,
  Tooltip,
  useToast,
  ValueContainer,
  withToast,
  withToastPropTypes,
};
