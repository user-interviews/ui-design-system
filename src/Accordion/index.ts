import Accordion from './Accordion';
import AccordionItem from './AccordionItem';
import AccordionToggle from './AccordionToggle';
import AccordionCollapse from './AccordionCollapse';

export {
  Accordion,
  AccordionItem,
  AccordionToggle,
  AccordionCollapse,
};
