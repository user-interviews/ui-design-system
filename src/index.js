import {
  Accordion,
  AccordionCollapse,
  AccordionItem,
  AccordionToggle,
} from 'src/Accordion';
import { Alert, MessageTypes } from 'src/Alert';
import Button from 'src/Button';
import Avatar from 'src/Avatar';
import Card, { CardSizes } from 'src/Card';
import CardContainer from 'src/CardContainer';
import CheckboxButton, { CHECKED_STATES } from 'src/CheckboxButton';
import CheckboxButtonGroup from 'src/CheckboxButtonGroup';
import { ORIENTATIONS as BUTTON_GROUP_ORIENTATIONS } from 'src/ControlButtonGroup';
import CopyToClipboard from 'src/CopyToClipboard';
import CopyToClipboardButton from 'src/CopyToClipboardButton';
import {
  Dropdown,
  DropdownToggle,
  DropdownItem,
  DropdownMenu,
} from 'src/Dropdown';
import FadeTransition from 'src/FadeTransition';
import Form from 'src/Form';
import FormControlLabel from 'src/FormControlLabel';
import FormGroup from 'src/FormGroup';
import IconCell from 'src/IconCell';
import Input from 'src/Input';
import InputLabel from 'src/InputLabel';
import InputLegend from 'src/InputLegend';
import LoadingOverlay from 'src/LoadingOverlay';
import {
  Modal,
  MODAL_SIZES,
  ModalBody,
  ModalFooter,
  ModalHeader,
} from 'src/Modal';
import { OverlayTrigger, OVERLAY_TRIGGER_PLACEMENT } from 'src/OverlayTrigger';
import { Pill, Pills, PILL_COLORS } from 'src/Pill';
import { Popover, PopoverBody } from 'src/Popover';
import Popper from 'src/Popper';
import ProfileCell from 'src/ProfileCell';
import RadioButton from 'src/RadioButton';
import RadioButtonGroup from 'src/RadioButtonGroup';
import {
  AsyncSelect,
  AsyncCreatableSelect,
  CreatableSelect,
  Option,
  SELECT_SIZES,
  SelectComponents,
  SingleSelect,
  ValueContainer,
} from 'src/Select';
import { colors as COLORS } from 'src/Styles';
import { Step, Steps } from 'src/Steps';
import {
  Tab,
  Tabs,
} from 'src/Tabs';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
} from 'src/Table';
import {
  Toast,
  useToast,
  withToast,
  withToastPropTypes,
} from 'src/Toast';
import { ToggleInput } from 'src/ToggleInput';
import Tooltip from 'src/Tooltip';

export {
  Accordion,
  AccordionCollapse,
  AccordionItem,
  AccordionToggle,
  Alert,
  Avatar,
  AsyncSelect,
  AsyncCreatableSelect,
  Button,
  BUTTON_GROUP_ORIENTATIONS,
  Card,
  CardContainer,
  CardSizes,
  CheckboxButton,
  CheckboxButtonGroup,
  CHECKED_STATES,
  COLORS,
  CopyToClipboard,
  CopyToClipboardButton,
  CreatableSelect,
  Dropdown,
  DropdownToggle,
  DropdownItem,
  DropdownMenu,
  FadeTransition,
  Form,
  FormControlLabel,
  FormGroup,
  IconCell,
  Input,
  InputLabel,
  InputLegend,
  LoadingOverlay,
  MessageTypes,
  Modal,
  MODAL_SIZES,
  ModalBody,
  ModalFooter,
  ModalHeader,
  Option,
  OverlayTrigger,
  OVERLAY_TRIGGER_PLACEMENT,
  Pill,
  Pills,
  PILL_COLORS,
  Popover,
  PopoverBody,
  Popper,
  ProfileCell,
  RadioButton,
  RadioButtonGroup,
  SELECT_SIZES,
  SelectComponents,
  SingleSelect,
  Step,
  Steps,
  Tab,
  Tabs,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
  Toast,
  ToggleInput,
  Tooltip,
  useToast,
  ValueContainer,
  withToast,
  withToastPropTypes,
};
