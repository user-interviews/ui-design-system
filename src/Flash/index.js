import { MessageTypes } from './FlashMessage';
import useFlash from './useFlash';
import withFlash, { withFlashPropTypes } from './withFlash';

export {
  MessageTypes,
  useFlash,
  withFlash,
  withFlashPropTypes,
};
