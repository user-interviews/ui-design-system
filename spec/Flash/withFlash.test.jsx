import React, { Component } from 'react';
import { act } from '@testing-library/react-hooks';
import { create } from 'react-test-renderer';

import withFlash from '../../src/Flash/withFlash';
import { MessageTypes } from '../../src/Flash';

jest.mock('react-transition-group', () => (
  {
    CSSTransition: ({ children }) => children,
    TransitionGroup: ({ children }) => children,
  }
));

// eslint-disable-next-line react/prefer-stateless-function
class WrappedComponent extends Component {
  render() {
    return <div />;
  }
}

describe('test withFlash', () => {
  test('it can create a new flash message', async () => {
    const newMessage = 'This is just a test...';
    const ComponentWithFlash = withFlash(WrappedComponent);
    const flash = create(<ComponentWithFlash />);
    const component = flash.root.findByType(WrappedComponent);


    act(() => {
      component.props.setFlashMessage(MessageTypes.SUCCESS, newMessage);
    });

    expect(flash).toMatchSnapshot();
  });
});
