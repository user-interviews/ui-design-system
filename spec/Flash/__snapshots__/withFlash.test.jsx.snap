// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test withFlash it can create a new flash message 1`] = `
Array [
  <div
    className="alert alert-success"
  >
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      Ã—
    </button>
    This is just a test...
  </div>,
  <div />,
]
`;
